src/main/java/org/opentripplanner/model/modes/TransitModeConfiguration.java

```diff
@@ -0,0 +1,78 @@
+package org.opentripplanner.model.modes;
+
+import org.opentripplanner.standalone.config.SubmodesConfig;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Contains all of the configured transit modes. The main modes are not configurable, and are
+ * accessible via a static method. This is instantiated by the SubmodesConfiguration graph builder
+ * module.
+ */
+public class TransitModeConfiguration {
+
+  private final Set<TransitMode> configuredTransitModes;
+
+  private static Map<TransitMainMode, TransitMode> mainTransitModes = Arrays
```

Comment:
The word _main_ is used in two different meanings here - witch is confusing. Is there a better name for this variable?

Reply:
Instead of main here maybe: default, plain, bare, basic can be used

Reply:
As with the other comment about main/submode naming, I think we should discuss this in an OTP meeting.